plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -103,11 +104,13 @@ func (d *defaultServicePlan) Admit(a admission.Attributes) error {
 	// the order changes, we will need to rethink the
 	// implementation of this controller.
 
-	// implement field selector
-
-	// loop over all service plans accumulate into slice
-	plans, err := d.spLister.List(labels.Everything())
-	// TODO filter `plans` down to only those owned by `sc`.
+	// loop over all service plans for the specified classname and accumulate into slice
+	plans, err := d.getClusterServicePlansByClusterServiceClassName(sc.Name)
+	if err != nil {
+		msg := fmt.Sprintf("Error listing plans for service class %q - retry and specify desired ClusterServicePlan", instance.Spec.ExternalClusterServiceClassName)
```

Comment:
Can you please disambiguate the k8s and external names of the ClusterServiceClass?